subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/strategy/StaticVersionComparator.java

```diff
@@ -24,7 +24,7 @@
 
 class StaticVersionComparator implements Comparator<Version> {
     private static final Map<String, Integer> SPECIAL_MEANINGS =
-            ImmutableMap.of("dev", new Integer(-1), "rc", new Integer(1), "final", new Integer(2));
+            ImmutableMap.of("dev", -1, "rc", 1, "final", 2, "release", 3);
```

Comment:
I do not have a strong opinion here. Adam mentions in the post that currently "release" < "final", so it would be slightly more of a breaking change to switch this order. 

@wilkinsona if we kept with "release" < "final" (revert 7f106d9), would it adversely affect any of Spring's projects?